### PR TITLE
feat: move shared contract addresses to shared repo

### PR DESCRIPTION
**Pull request by cbachmeier**
We want to consolidate much of the repeated code in the UX and BE involved with launching on a new chain. This PR is to centralize a lot of the Uni contract addresses across chains in one place. 

I've also restructured how these contracts are stored, so instead of a series of repeated strings such as `SEPOLIA_V3_CORE_FACTORY_ADDRESSES` without any programmatic method to enforce consistency, there is a new datatype `ChainAddresses` whose fields correspond to the related contract. There is a map `CHAIN_TO_ADDRESSES_MAP` which will return all the `ChainAddresses` for a given `ChainId`. This helps simplify a lot of the helper `consts` such as `V3_CORE_FACTORY_ADDRESSES` which is now simply a reduce function across all `SUPPORTED_CHAINS`.

Last change primarily affecting UX is I've swapped `SupportedChainId` for `ChainId` and now `SUPPORTED_CHAINS` is an array which is the subset of all available `Chains` where Uni addresses have been deployed. This is the format used by the BE and is more robust than the previous sole `SupportedChainId`. This will change will likely require some small refactoring on the UX but will have us closer aligned to the BE.

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -1,25 +1,48 @@
...
   CELO_ALFAJORES = 44787,
-
+  GNOSIS = 100,
+  MOONBEAM = 1284,
   BNB = 56
 }
 
+export const SUPPORTED_CHAINS = [
+  ChainId.MAINNET,
+  ChainId.RINKEBY,
```
Comment (hensha256) [inline src/constants.ts:27]: you can get rid of some of these chains - theyre gone forever

Comment (hensha256) [reply src/constants.ts:27]: rinkeby, ropsten (i think on L2s too), and kovan as well - all test networks that dont exist anymore

Comment (cbachmeier) [reply src/constants.ts]: @lynnshaoyu I brought these chains over from the BE repo, can we deprecate across the org?

File: src/addresses.ts [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+const SEPOLIA_ADDRESSES: ChainAddresses = {
+  v3CoreFactoryAddress: '0x0227628f3F023bb0B980b67D528571c95c6DaC1c',
+  multicallAddress: '0xD7F33bCdb21b359c8ee6F0251d30E94832baAd07',
+  quoterAddress: '0xEd1f6473345F45b75F8179591dd5bA1888cf2FB3',
+  v3MigratorAddress: '0x729004182cF005CEC8Bd85df140094b6aCbe8b15',
+  nonfungiblePositionManagerAddress: '0x1238536071E1c677A632429e3655c799b22cDA52',
+  tickLensAddress: '0xd7f33bcdb21b359c8ee6f0251d30e94832baad07'
+}
+
+export const CHAIN_TO_ADDRESSES_MAP: Record<SupportedChainsType, ChainAddresses> = {
```
Comment (lynnshaoyu) [inline src/addresses.ts:122]: yay! this will make sor code referencing the addresses a lot simpler!

File: src/addresses.ts [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+  multicallAddress: string
+  quoterAddress: string
+  v3MigratorAddress?: string
+  nonfungiblePositionManagerAddress?: string
+  tickLensAddress?: string
+  swapRouter02Address?: string
+  v1MixedRouteQuoterAddress?: string
+}
+
+const DEFAULT_NETWORKS = [ChainId.MAINNET, ChainId.GOERLI, ChainId.SEPOLIA]
```
Comment (lynnshaoyu) [inline src/addresses.ts:16]: what makes a network default? slightly confused that sepolia is part of this when it doesn't use the DEFAULT_ADDRESSES unlike goerli and mainnet

Comment (cbachmeier) [reply src/addresses.ts:16]: good callout, it's mainly used in the `constructSameAddressMap` which I put Sepolia in as it shares a uni address, but this was a mistake as it doesn't share ens registrar for example. I will remove it from `DEFAULT_NETWORKS` and instead append it to the `UNI_ADDRESSES` map

File: src/constants.ts (hunk 1 above)
Comment (cbachmeier) [reply src/constants.ts]: confirmed offline that these can be removed

**Comment by cbachmeier** [comment]
Take a look at #59 to see how a new chain's addresses (Avalanche) are added

File: src/addresses.ts [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+}
+
+const DEFAULT_NETWORKS = [ChainId.MAINNET, ChainId.GOERLI]
+
+function constructSameAddressMap(address: string, additionalNetworks: ChainId[] = []): AddressMap {
+  return DEFAULT_NETWORKS.concat(additionalNetworks).reduce<AddressMap>((memo, chainId) => {
+    memo[chainId] = address
+    return memo
+  }, {})
+}
```
Comment (JFrankfurt) [inline src/addresses.ts:23]: I find the `DEFAULT_NETWORKS` piece a bit confusing personally--I know you've copied it from interface and divergent implementations would be a little annoying, but requiring explicit enumeration of the full set at runtime feels more predictable to me.

**Review by JFrankfurt — APPROVED** [review]

File: src/addresses.ts (hunk 4 above)
Comment (cbachmeier) [reply src/addresses.ts:23]: Do you think the added clarity is worth the tradeoff of the redundancy of having to copy&paste Mainnet and Goerli in several locations? Could we address this in a different way such as renaming `DEFAULT_NETWORKS`?

**Pull request merged**